### DISCUSSION TITLE
OperationBatch/OperationEmitter: split the sign and the inject actions

### DISCUSSION THREAD

**Discussion by bchevalier74** [ecadlabs/taquito#1129] [Ideas]
Hello,

currently, my understanding is that it's not possible with Taquito to sign an operation and retrieve the object without sending it to the blockchain. The function OperationEmitter.signAndInject gathers the 2 functionalities as the functions RpcContractProvider.transfer or OperationBatch.send. 

Is it possible to split the functions OperationEmitter.signAndInject, RpcContractProvider.transfer, OperationBatch.send in order to have for example the functions 
RpcContractProvider.prepareTransferOperation : create the Operation and sign it and return the created object
RpcContractProvider.sendOperation : send the operation to the blockchain.
RpcContractProvider.transfer would remain unchanged and would call RpcContractProvider.prepareTransferOperation and RpcContractProvider.sendOperation.

Currently, to have this functionality, I have to extend RpcClient in order to empty the function injectOperation and to get the operation in the preapplyOperations. It's more a workaround than a real solution.

Bruno CHEVALIER.

**Comment by roxaneletourneau** [comment]
Hi,
When using the contract API to send an operation to the blockchain, there is no visibility on the underlying steps: fees estimation, preparation of the operation in an RPC compliant format, forging, signing, simulation, and injection. These steps are intentionally abstracted to provide the user with an easy way to interact with the blockchain. However, like is the case here, sometimes a user may want to retrieve information from one of the particular underlying steps without getting to the operation injection.

At the moment, Taquito does not provide a way to "stop/pause" the process at a particular step in order to retrieve the prepared object. There are lower APIs that allow more flexibility but required more work from the user. For example, here is how a contract method can be called using "manual steps": https://github.com/ecadlabs/taquito/blob/master/integration-tests/call-contract-method-manual-steps.spec.ts.
We are conscient that it is not evident to prepare the operation into the RPC expected format. 
So, we are currently designing a solution, similar to what you suggested with the "prepareTransferOperation" in order to make these use cases more convivial. The work can be tracked in this issue: https://github.com/ecadlabs/taquito/issues/432 (but note that the solution will be different from the one suggested in the ticket, no breaking change will be introduced).
